### PR TITLE
BGE: Allow only selected fields for donation matching options

### PR DESCRIPTION
**Pull request by psadie**
- Update to only display matching options that were previously selected in the "Field Selection" step of the config wizard.
- Update to custom label as per conversation with docs team [chatter](https://foundation.lightning.force.com/lightning/r/0D51E00004kUa9ZSAS/view) 

# Critical Changes

# Changes

# Issues Closed

# New Metadata

# Deleted Metadata

### CODE REVIEW

**Comment by psadie** [comment]
**lurch: attach W-031175

**Comment by LurchTheButler** [comment]
Tracking <a href="https://foundation.my.salesforce.com/a2x1E000007fRKrQAM" target="_blank">W-031175</a>

File: src/aura/BGE_ConfigurationWizard/BGE_ConfigurationWizardHelper.js [hunk 1]
```diff
@@ -442,12 +443,17 @@
...
+                    label: currentField.label,
+                    value: currentField.name
+                });
             });
 
             batchFieldOptions.fieldGroups.push(currentFieldGroup);
 
         });
         component.set('v.batchFieldOptions', batchFieldOptions);
+        component.set('v.batchInfo.donationMatchingOptions', donationMatchingOptions);
```
Comment (ptewson-sfdo) [inline src/aura/BGE_ConfigurationWizard/BGE_ConfigurationWizardHelper.js]: How were we setting donationMatchingOptions previously? Was it just getting a copy of the batch fields somewhere?

Comment (ptewson-sfdo) [reply src/aura/BGE_ConfigurationWizard/BGE_ConfigurationWizardHelper.js]: (UPDATE: I think I found it, it's getting all the fields from getRelevantFields() in the apex controller and then previous to this work, just sticking with that list)

Comment (psadie) [reply src/aura/BGE_ConfigurationWizard/BGE_ConfigurationWizardHelper.js]: That's what I've just found as well

File: src/labels/CustomLabels.labels [hunk 2]
```diff
@@ -2600,7 +2600,7 @@ To check your GAU Allocation settings, go to Donations | GAU Allocations.</value
         <language>en_US</language>
         <protected>true</protected>
         <shortDescription>Used as field help text since help text is not upgradeable.</shortDescription>
-        <value>Specifies which fields to match by when trying to match Donations in batch records against existing Opportunties and Payments.</value>
+        <value>You can match against any of the Opportunity or Payment fields that you chose to include in this batch.</value>
```
Comment (ptewson-sfdo) [inline src/labels/CustomLabels.labels]: Did we consult with Melissa/documention team for this language?

Comment (psadie) [reply src/labels/CustomLabels.labels]: Sure did, thread is over here https://foundation.lightning.force.com/lightning/r/0D51E00004kUa9ZSAS/view

File: src/aura/BGE_ConfigurationWizard/BGE_ConfigurationWizardHelper.js (hunk 1 above)
Comment (bethbrains) [reply src/aura/BGE_ConfigurationWizard/BGE_ConfigurationWizardHelper.js]: If we're setting it here only, and we always call this function in the course of the wizard, we don't need to pass back anything from apex, so we can actually remove donationMatchingOptions from the model and remove the whole method that sets it in Apex.

File: src/labels/CustomLabels.labels (hunk 2 above)
Comment (kslewis) [inline src/labels/CustomLabels.labels]: 👏

File: src/aura/BGE_ConfigurationWizard/BGE_ConfigurationWizardHelper.js [hunk 3]
```diff
@@ -451,8 +451,17 @@
...
 
         });
+
+        let selectedMatchingRule = component.get('v.batchInfo.donationMatchingRule');
+        let donationMatchingOptionValues = donationMatchingOptions.map(function(option) { return option.value });
+        // Filter out any selected matching rules that aren't selected as available matching fields
+        selectedMatchingRule = selectedMatchingRule.filter(function (selectedRule) {
+            return (donationMatchingOptionValues.indexOf(selectedRule) >= 0);
+        });
+
```
Comment (psadie) [inline src/aura/BGE_ConfigurationWizard/BGE_ConfigurationWizardHelper.js]: @ptewson-sfdo @kenslewis @bethbrains - this is the commit I mentioned during stand up today.

File: src/aura/BGE_ConfigurationWizard/BGE_ConfigurationWizardHelper.js [hunk 4]
```diff
@@ -442,12 +442,26 @@
...
+                    value: currentField.name.toLowerCase()
+                });
             });
 
             batchFieldOptions.fieldGroups.push(currentFieldGroup);
 
         });
+
+        let selectedMatchingRule = component.get('v.batchInfo.donationMatchingRule');
+        let donationMatchingOptionValues = donationMatchingOptions.map(function(option) { return option.value });
```
Comment (ptewson-sfdo) [inline src/aura/BGE_ConfigurationWizard/BGE_ConfigurationWizardHelper.js]: Suggesting breaking this into separate lines for parens and code

File: src/aura/BGE_ConfigurationWizard/BGE_ConfigurationWizardHelper.js [hunk 5]
```diff
@@ -453,7 +453,9 @@
         });
 
         let selectedMatchingRule = component.get('v.batchInfo.donationMatchingRule');
-        let donationMatchingOptionValues = donationMatchingOptions.map(function(option) { return option.value });
+        let donationMatchingOptionValues = donationMatchingOptions.map(function (option) {
+            return option.value;
+        });
         // Filter out any selected matching rules that aren't selected as available matching fields
```
Comment (psadie) [inline src/aura/BGE_ConfigurationWizard/BGE_ConfigurationWizardHelper.js]: @ptewson-sfdo - Broke those out into newlines as suggested :)

File: src/aura/BGE_ConfigurationWizard/BGE_ConfigurationWizardHelper.js (hunk 3 above)
Comment (ptewson-sfdo) [reply src/aura/BGE_ConfigurationWizard/BGE_ConfigurationWizardHelper.js]: This approach looks good - nice work turning that around so quickly

**Pull request merged**